src/v/pandaproxy/auth_utils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "config/rest_authn_endpoint.h"
+#include "pandaproxy/types.h"
+#include "redpanda/request_auth.h"
+
+#include <seastar/http/request.hh>
+
+#include <vector>
+
+namespace pandaproxy {
+namespace {
+credential_t validate_credentials(
```

Comment:
could you write a comment on this method? the name `validate_credentials` suggests to me that the return should be `void` or possibly `bool`. so i think it isn't quite clear what the semantics are here, especially when a default-initialized user is returned.